File: src/model/Figure.java
package model;

import java.awt.Point;

import javax.swing.JLabel;

import model.Constants.FigureType;
import model.Constants.PlayerColor;

public class Figure {
	
	private FigureType figureType;
	private PlayerColor playerColor;
	private Point point;
	private JLabel figureImage;
	
	public Figure() {
		super();
	}
	
	public Figure(FigureType figureType, PlayerColor playerColor, Point point, JLabel figureImage) {
		super();
		this.figureType = figureType;
		this.playerColor = playerColor;
		this.point = point;
		this.figureImage = figureImage;
	}

	public FigureType getFigureType() {
		return figureType;
	}

	public void setFigureType(FigureType figureType) {
		this.figureType = figureType;
	}

	public PlayerColor getPlayerColor() {
		return playerColor;
	}

	public void setPlayerColor(PlayerColor playerColor) {
		this.playerColor = playerColor;
	}

	public Point getPoint() {
		return point;
	}

	public void setPoint(Point point) {
		this.point = point;
	}
	
	public JLabel getFigureImage() {
		return figureImage;
	}

	public void setFigureImage(JLabel figureImage) {
		this.figureImage = figureImage;
	}
}

File: src/model/Figure.scala
package model

import javax.swing.JLabel
import model.FigureType._
import model.PlayerColor._

case class Figure(var figureType : FigureType, color : PlayerColor, var x : Int, var y: Int, 
	var figureImage : JLabel, var moved : Boolean = false) {

  def getFigureImage: JLabel = figureImage
  def setFigureImage(figureImage: JLabel): Unit = this.figureImage = figureImage
  def getPoint: (Int, Int) = (x, y)
  def getX: Int = x
  def getY: Int = y
  def setPoint(_x: Int, _y: Int): Unit = {
    this.x = _x
    this.y = _y
  }
  def XY() : (Int, Int) = (x, y)
  def getType(): FigureType = figureType
  def getColor(): PlayerColor = color
	def hasMoved(): Boolean = moved
  def setMoved(moved : Boolean) : Unit = this.moved = moved
	def setFigureType(figureType : FigureType) : Unit = this.figureType = figureType
}
